### PR TITLE
Update project search to allow extending the searchable items when ne…

### PR DESCRIPTION
**Pull request by mellisio**

### CODE REVIEW

File: app/models/concerns/project_search.rb [hunk 1]
```diff
@@ -51,6 +52,14 @@ def exact_name
       name
     end
 
+    def extra_searchable_names
+      if platform == "Maven"
+        name.split(":")
+      else
+        [name]
```
Comment (katzj) [inline app/models/concerns/project_search.rb]: For other platforms, can we get by with just an empty list for now?

File: app/models/concerns/project_search.rb [hunk 2]
```diff
@@ -51,6 +52,14 @@ def exact_name
       name
     end
 
+    def extra_searchable_names
+      if platform == "Maven"
+        name.split(":")
```
Comment (katzj) [inline app/models/concerns/project_search.rb]: So this will let searches return both group id and artifact id? That sounds reasonable I think

Comment (mellisio) [reply app/models/concerns/project_search.rb]: yes.

**Comment by katzj** [comment]
It looks like we can add the field to the existing index with something like
```bash
curl -X PUT http://localhost:9200/projects/_mapping/project -H "Content-Type: application/json" -d'{"properties": {"extra_searchable_names": {"type": "string"}}}'
```

Then it'll get updated as projects update. We can force a resync of the maven stuff to get them which is the primary set we know about for now.

**Pull request merged**